WalletWasabi.Tests/UnitTests/WabiSabi/Client/CoinJoinCoinSelectionTests.cs

```diff
@@ -151,6 +151,35 @@ public void SelectTwoNonPrivateCoinsFromTwoCoinsSetOfCoinsConsolidationMode()
 		Assert.Equal(2, coins.Count);
 	}
 
+	[Fact]
+	public void DoNotSelectCoinsWithBigAnonymityLoss()
+	{
+		// This test ensures that we do not select coins whose anonymity could be lowered a lot
+		const int AnonymitySet = 10;
```

Comment:
Yes, it would make sense. Nevertheless, it is not easy to generate such test vectors, because the coin selection algorithm is very complex and doesn't behave deterministically.